docs/db/changelog/changesets/powerauth-webflow/1.5.x/20230510-spring-authorization-server-1.5.0.xml

```diff
@@ -0,0 +1,184 @@
+                <constraints nullable="false" />
+            </column>
+            <column name="client_secret" type="varchar(200)"/>
+            <column name="client_secret_expires_at" type="timestamp"/>
+            <column name="client_name" type="varchar(100)">
+                <constraints nullable="false" />
+            </column>
+            <column name="client_authentication_methods" type="varchar(1000)">
+                <constraints nullable="false" />
+            </column>
+            <column name="authorization_grant_types" type="varchar(1000)">
+                <constraints nullable="false" />
+            </column>
+            <column name="redirect_uris" type="varchar(1000)"/>
+            <column name="post_logout_redirect_uris" type="varchar(1000)"/>
+            <column name="scopes" type="varchar(1000)">
+                <constraints nullable="false" />
+            </column>
+            <column name="client_settings" type="varchar(2000)">
+                <constraints nullable="false" />
+            </column>
+            <column name="token_settings" type="varchar(2000)">
+                <constraints nullable="false" />
+            </column>
+        </createTable>
+        <!-- no rollback -->
+        <rollback />
+    </changeSet>
+
+    <!-- Migration to Spring authorization server: table oauth2_authorization_consent -->
+    <changeSet id="3" logicalFilePath="powerauth-webflow/1.5.x/20230510-spring-authorization-server-1.5.0.xml" author="Roman Strobl">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <tableExists tableName="oauth2_authorization_consent"/>
+            </not>
+        </preConditions>
+        <comment>Create a new table oauth2_authorization_consent</comment>
+        <createTable tableName="oauth2_authorization_consent">
+            <column name="registered_client_id" type="varchar(100)">
+                <constraints primaryKey="true" primaryKeyName="pk_consent" nullable="false"/>
+            </column>
+            <column name="principal_name" type="varchar(200)">
+                <constraints primaryKey="true" primaryKeyName="pk_consent" nullable="false" />
+            </column>
+            <column name="authorities" type="varchar(1000)">
+                <constraints nullable="false" />
+            </column>
+        </createTable>
+        <!-- no rollback -->
+        <rollback />
+    </changeSet>
+
+    <!-- Migration to Spring authorization server: drop table oauth_client_details -->
+    <changeSet id="4" logicalFilePath="powerauth-webflow/1.5.x/20230510-spring-authorization-server-1.5.0.xml" author="Roman Strobl">
+        <preConditions onFail="MARK_RAN">
+            <tableExists tableName="oauth_client_details"/>
+        </preConditions>
+        <comment>Drop table oauth_client_details</comment>
+        <dropTable tableName="oauth_client_details"/>
+        <!-- no rollback -->
```

Comment:
🤔 We do not work much with rollbacks yet, but to be perfect, rollback for `drop` could be create table from the init script. But is a quite philosophical question.

Reply:
Yes, I can prepare the rollback script.